pori_python/ipr/connection.py

```diff
@@ -101,6 +101,17 @@ def upload_report(
             # if async is used, the response for reports-async contains either 'jobStatus'
             # or 'report'. jobStatus is no longer available once the report is successfully
             # uploaded.
+
+            projects = self.get("project")
+            project_names = [item['name'] for item in projects]
+
+            # if project is not exist, create one
+            if content['project'] not in project_names:
```

Comment:
add a logline here saying something like 'Project not found - attempting to create project' so that the user knows this is happening; if it fails because user doesn't have permission to create project also log that

Reply:
'Project creation failed due to {error}' probably user permissions